api/server.js

```diff
@@ -185,11 +183,19 @@ app.post("/bookshelves/remove", async (req, res) => {
  * Note: All parameters are optional (Should we require at least 1 parameter?)
  * If no books match search criteria... return an empty JSON Object
  */
-app.get("/books/search/:title/:author/:genre/:offset/:resultLength", async function(req, res) {
+app.get("/books/search", async function(req, res) {
 	try {
-		
+		res.type("json");
+		let offset = req.query.offset ? req.query.offset : 0;
+		let resultLength = req.query.resultLength ? req.query.resultLength : 10;
+		let books = await getMatchingBooks(req.query);
+		res.status(SUCCESS_CODE).json({
+			remainingBooksInSearch : books.slice(offset + resultLength).length,
+			books : books.slice(offset, offset + resultLength)
```

Comment:
Bug: Including both `resultLength` and `offset` returns a list of books that is greater than the `resultLength`.
Debugging results: `offset` and `resultLength` are read as strings by nodejs right now. When we add them, they are added as strings ("1" + "2" = "12")

I'll have time to fix this Saturday but you are welcome to do so yourself if you like beforehand.

Reply:
Ah good catch. It should be fixed. Do we need to do error handling for invalid inputs (ie. front-end passes in a word instead of a number to "offset")?

Reply:
I think we should at some point but not now. I want to get this branch merged ASAP